sql/parser/sql.y

```diff
@@ -807,6 +814,51 @@ explain_option_arg:
 // | create_index_stmt {}
 // | create_view_stmt {}
 
+// GRANT privileges ON privilege_target TO grantee_list
+grant_stmt:
+  GRANT privileges ON privilege_target TO grantee_list
+  {
+    $$ = &Grant{Privileges: $2, Target: Name($4), Grantees: $6}
+  }
+
+// TODO(marc): permissions are currently at the database level.
+// Adjust this to allow tables (including db.* expressions).
+privilege_target:
+  DATABASE name
+  {
+    $$ = $2
+  }
+
+// TODO(marc): this needs to change once privileges are reserved keywords.
+privileges:
+  ALL
+  {
+    $$ = []string{"ALL"}
+  }
+  | privilege_list { }
+
+privilege_list:
+  name
+  {
+    $$ = []string{$1}
+  }
+  | privilege_list ',' name
+  {
+    $$ = append($1, $3)
+  }
```

Comment:
I don't think it is worth following the postgres grammar precisely here unless we want to follow their privilege model as well (which I don't think we do).